File: GlueCustomConnectors/development/Spark/glue-3.0/tpcds-custom-connector-for-glue3.0/jobvalidation/pyspark/glue_job_validation_datasource_test.py
#  Copyright 2016-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
#  SPDX-License-Identifier: MIT-0

from pyspark.context import SparkContext
from awsglue.context import GlueContext


glue_context = GlueContext(SparkContext())
logger = glue_context.get_logger()

connection_options = {
    "table": "customer",
    "scale": "1",
    "numPartitions": "1",
    "connectionName" : "GlueTPCDSConnection"
}

# read data from data source
datasource0 = glue_context.create_dynamic_frame_from_options(
    connection_type="marketplace.spark",
    connection_options=connection_options)

# validate data reading and row count
expected_count = 100000
result_count = datasource0.count()
assert result_count == expected_count
logger.info(f'Expected record count: {expected_count}, result record count: {result_count}')

File: GlueCustomConnectors/development/Spark/glue-3.0/tpcds-custom-connector-for-glue3.0/jobvalidation/scala/GlueJobValidationDataSourceTest.scala
/*
 * Copyright 2016-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

import com.amazonaws.services.glue.GlueContext
import com.amazonaws.services.glue.errors.CallSite
import com.amazonaws.services.glue.util.GlueArgParser
import com.amazonaws.services.glue.util.Job
import com.amazonaws.services.glue.util.JsonOptions
import org.apache.spark.SparkContext
import scala.collection.JavaConverters._

object GlueJobValidationDataSourceTest {
  def main(sysArgs: Array[String]) {
    val spark: SparkContext = new SparkContext()
    val glueContext: GlueContext = new GlueContext(spark)

    val optionsMap = Map(
      "table" -> "customer",
      "scale" -> "1",
      "numPartitions" -> "1",
      "connectionName" -> "GlueTPCDSConnection"
    )

    // create DataSource
    val customSource = glueContext.getSource(
      connectionType = "marketplace.spark",
      connectionOptions = JsonOptions(optionsMap),
      transformationContext = "customSource")
    val dyf = customSource.getDynamicFrame()

    // verify data
    val expectedRowCount = 100000
    assert(dyf.count == expectedRowCount)
  }
}